src/ray/raylet/node_manager.cc

```diff
@@ -567,9 +567,25 @@ void NodeManager::HandleReleaseUnusedBundles(
                        bundle_id.bundle_index()));
   }
 
-  // TODO(ffbin): Kill all workers that are currently associated with the unused bundles.
-  // At present, the worker does not have a bundle ID, so we cannot filter out the workers
-  // used by unused bundles. We will solve it in next pr.
+  // Kill all workers that are currently associated with the unused bundles.
+  for (const auto &worker_it : leased_workers_) {
+    auto &worker = worker_it.second;
+    if (0 == in_use_bundles.count(worker->GetBundleId())) {
+      RAY_LOG(DEBUG)
+          << "Destroying worker since its bundle was unused. Placement group id: "
+          << worker->GetBundleId().first
+          << ", bundle index: " << worker->GetBundleId().second
+          << ", task id: " << worker->GetAssignedTaskId()
+          << ", actor id: " << worker->GetActorId()
+          << ", worker id: " << worker->WorkerId();
+      // We should disconnect the client first. Otherwise, we'll remove bundle resources
+      // before actual resources are returned. Subsequent disconnect request that comes
+      // due to worker dead will be ignored.
+      ProcessDisconnectClientMessage(worker->Connection(), /* intentional exit */ true);
```

Comment:
If this was copied from our `CancleBundleResource`, can we make this a private method?

Reply:
Yeah, i will fix it, thanks.